Code/Engine/AudioManager.cpp

```diff
@@ -0,0 +1,103 @@
+#include "sadpch.h"
+#include "AudioManager.h"
+
+std::list<sad::AudioFile> sad::AudioManager::m_AudioFiles;
+
+void sad::AudioManager::loadAudioFile(const std::string& filename, const std::string& path, int volume, AudioType type)
+{
+    AudioFile audioFile;
+    audioFile.filename = filename;
+    audioFile.path = path.c_str();
+    audioFile.type = type;
+    audioFile.volume = volume;
+
+    core::Log(ELogType::Debug, "Audio Path: {}", path.c_str()); // Debug Path
+
+    if (Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, 2, 512) < 0) // Check if audio file can be opened
+    {
+        core::Log(ELogType::Debug, "Couldn't open audio sample: {}", Mix_GetError());
+    }
+    else
+    {
+        if (type == MP3)
+        {
+            audioFile.mixMusic = Mix_LoadMUS(path.c_str());
+            if (audioFile.mixMusic != NULL)
+            {
+                core::Log(ELogType::Debug, "Audio sample loaded!");
+                Mix_VolumeMusic(volume);
+
+                m_AudioFiles.push_back(audioFile);
+            }
+            else
+            {
+                core::Log(ELogType::Debug, "Couldn't load audio sample: {}", Mix_GetError());
+            }
+        }
+
+        if (type == WAV)
+        {
+            audioFile.mixChunk = Mix_LoadWAV(path.c_str());
+            if (audioFile.mixChunk != NULL)
+            {
+                core::Log(ELogType::Debug, "Audio sample loaded!");
+                Mix_VolumeChunk(audioFile.mixChunk, volume);
+                
+                m_AudioFiles.push_back(audioFile);
+            }
+            else
+            {
+                core::Log(ELogType::Debug, "Couldn't load audio sample: {}", Mix_GetError());
+            }
+        }
```

Comment:
We can probably make this super efficient and clean by leveraging the new ResourceManager.

I've already made an `AudioResource.cpp` that will have a handful of these settings automatically generated such as file name, path, and extension. What we can do is move the audio-file specific initialization code (such as mixing the volume chunks) to there and remove the static list of audio files in the audio manager.

In that sense, the `AudioManager` could just take a single `AudioResource` as most of the parameters (removing the static list and need for loops 🆒 ) here (ie for `SetVolume` or `PlaySFX`) and just play what's already been created for the audio resource by the resource manager.

Converted methods with `AudioResource` might look something like...
```
void sad::AudioManager::PlaySFX(const AudioResource& audioResource)
{
    if (Mix_Playing(-1) == 0)
    {
        Mix_PlayChannel(-1, audioResource.MixChunk, 0);
    }
} 
```

Playing an audio resource (in a script - when it comes 😄 ) or 'on the fly' would end up looking like this...
```
AudioResource* audioResource = ResourceManager::GetResource<AudioResource>("Code_Arabic_Nokia.wav");
AudioManager::GetInstance().PlaySFX(audioResource);
```